gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobStatus.java

```diff
@@ -0,0 +1,110 @@
+package com.google.gcloud.bigquery;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google BigQuery Job status. Objects of this class can be examined when polling an asynchronous
+ * job to see if the job completed.
+ */
+public class JobStatus implements Serializable {
+
+  private static final long serialVersionUID = -714976456815445365L;
+  public enum State {
+    PENDING, RUNNING, DONE
+  }
+
+  private final State state;
+  private final BigQueryError error;
+  private final List<BigQueryError> allErrors;
+
+  JobStatus(State state) {
+    this.state = state;
+    this.error = null;
+    this.allErrors = null;
+  }
+
+  JobStatus(State state, BigQueryError error, List<BigQueryError> allErrors) {
+    this.state = state;
+    this.error = error;
+    this.allErrors = allErrors != null ? ImmutableList.copyOf(allErrors) : null;
+  }
+
+  /**
+   * Returns the state of the job.
+   */
+  public State state() {
+    return state;
+  }
+
+  /**
+   * Returns the final error result of the job. If present, indicates that the job has completed
+   * and was unsuccessful.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/troubleshooting-errors">
+   *     Troubleshooting Errors</a>
+   */
+  public BigQueryError error() {
+    return error;
+  }
+
+  /**
+   * Returns all errors encountered during the running of the job. Errors here do not necessarily
+   * mean that the job has completed or was unsuccessful.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/troubleshooting-errors">
+   *     Troubleshooting Errors</a>
+   */
+  public List<BigQueryError> allErrors() {
```

Comment:
`'runErrors` or `executionErrors`,..?

I think "all" as a prefix is confusing considering that it may not include the completion error.